### PR TITLE
feat: making donut having a loading state

### LINKED ISSUE #5837: ContainerStatistics displaying `-1` before displaying real values
### Is your enhancement related to a problem? Please describe

When opening a container info page, the values for RAM and CPU can take some time to be displayed, (3-4 seconds), we might want to show something like `Collection` or else instead of `-1`

image

### Describe the solution you'd like

Either a loading animation, or a default message

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Adding a loading state for the `Donut.svelte` component. This is backward compatible, as the loading state is `false` by default.

> This PR only add the support to the Donut, no component are using it for now, it will be in a follow up PR

### Screenshot / video of UI

See comments 

### What issues does this PR fix or reference?

Related to https://github.com/containers/podman-desktop/issues/5837

### How to test this PR?

- [x] Unit tests has been added

### CODE REVIEW

File: packages/renderer/src/lib/donut/Donut.spec.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
```
Comment (axel7083) [inline packages/renderer/src/lib/donut/Donut.spec.ts]:
```suggestion
 * Copyright (C) 2024 Red Hat, Inc.
```

**Comment by deboer-tim** [comment]
I think we should still show the title during loading, otherwise I wonder why there are two spinners and have to hover over to see their purpose.

**Comment by axel7083** [comment]
> I think we should still show the title during loading, otherwise I wonder why there are two spinners and have to hover over to see their purpose.

@deboer-tim Here is an updated version displaying the provided `value` even when loading

> This mean the parent is responsible of displaying an appropriate value when loading.

https://github.com/containers/podman-desktop/assets/42176370/71983da6-5e7b-4929-91b4-bc9bb136d206

File: packages/renderer/src/lib/donut/Donut.svelte [hunk 2]
```diff
@@ -21,21 +22,40 @@ function describeArc(radius: number, endAngle: number) {
...
   <svg viewBox="-4 -4 {size + 8} {size + 8}" height="{size}" width="{size}">
-    <circle fill="none" class="stroke-charcoal-300" stroke-width="1" r="{size / 2}" cx="{size / 2}" cy="{size / 2}"
-    ></circle>
-    <path
-      fill="none"
-      class="{stroke}"
-      stroke-width="3.5"
-      d="{describeArc(size / 2, (percent * 360) / 100)}"
-      data-testid="arc"></path>
+    <g class:animate-spin="{loading}" class:origin-[50%_50%]="{loading}" style="transform-box: fill-box;">
```
Comment (axel7083) [inline packages/renderer/src/lib/donut/Donut.svelte:54]: the `transform-box: fill-box;` is not yet supported by tailwinds, ref https://github.com/tailwindlabs/tailwindcss/pull/10982

**Comment by deboer-tim** [comment]
Note that I was just suggesting the title was shown while loading, not value. :)

This got me thinking though: why introduce a new exported loading property, where we could end up with cases like 'loading but value shown' or 'not loading but no value' vs just making the loading animation appear whenever the value isn't set (or perhaps is set to an invalid value)?

**Comment by axel7083** [comment]
> Note that I was just suggesting the title was shown while loading, not value. :)
> 
> This got me thinking though: why introduce a new exported loading property, where we could end up with cases like 'loading but value shown' or 'not loading but no value' vs just making the loading animation appear whenever the value isn't set (or perhaps is set to an invalid value)?

Because I wanted to be backward compatible as some elements are using the Donut component. It is honestly pretty complicated for nothing, for example, why the donut is responsible of building the tooltip (`${percent}% ${title} usage`), it is not generic, nor easily reusable. 

My personal thought:

(1) It should act as a progress bar
(2) It should have 4 properties `label: string`, `value: number`, `loading: boolean`, `color: string`
- label should be used in the tooltip + for the text
- the value should be used for the progress
- loading to make the animation or not
- color for the progress indicator

Then, a wrapper component could be created to be used for resource usage etc.

**Comment by deboer-tim** [comment]
Donut isn't external API, so we only need to make sure we don't break existing places where it's used inside Podman Desktop - but we can always fix those references when we're improving the component. +1 to improving the way the tooltip works or changing title to label.

The original issue didn't talk about using Donut as a general progress widget, and our designs normally use Spinner or LinearProgress for that. Can you point me at where/how it would be used? IMHO some kind of 'if (!value) then loading = true' would be a good improvement to make either way for most of our current use.

**Comment by axel7083** [comment]
Closing for now, might want to come back later -

**Pull request closed without merging**